packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -282,6 +355,46 @@ const EDiscovery = WebexPlugin.extend({
       });
   },
 
+  @waitForValue('@')
+  @oneFlight({keyFactory: (reportUrl, spaceId) => String(reportUrl + spaceId)})
+  /**
+   * Retrieves information for a specific conversation relevant to a specified report
+   * @param {String} reportUrl - full destination address (including report id parameter) for the http request being sent
+   * e.g. 'http://ediscovery-intb.wbx2.com/ediscovery/api/v1/reports/3b10e625-2bd5-4efa-b866-58d6c93c505c'
+   * @param {UUID} spaceId - Id of the conversation being requested
+   * @param {Object} options - optional parameters for this method
+   * @param {number} options.timeoutMs - connection timeout in milliseconds, defaults to 30s
+   * @returns {Promise<ResponseEntity<ContentSummary>>} Http response containing the specified content summary
+   */
+  getRemoteContentSummaryBySpaceId(reportUrl, spaceId, options) {
+    if (!reportUrl || !spaceId) {
+      throw Error('Undefined parameter(s)');
+    }
+
+    const reportId = this._getReportIdFromUrl(reportUrl);
+
+    if (!reportId) {
+      throw Error('Report url does not contain a report id');
+    }
+
+    // use spread operator to set default options
+    options = {...this.config.defaultOptions, ...options};
```

Comment:
It seems like this block of code is repeated between the appended functions, While it's not required, would it be possible to make a helper function that handles the "setup" part of these methods? Possibly wrap the `reportUrl` and `options` values into a single method that can be called by the others?

Reply:
It would be nice to remove some of the duplication Tim, but I'd be concerned it would look inconsistent if I changed it for those 5 functions that this block of code appears in and leaving the other 8 as they are. There are a variety of input params across the functions here, I'd rather not create too many helper functions which might end up making it harder to read.